### DISCUSSION TITLE
Is it possible to use Stimulus or other web components libraries?

### DISCUSSION THREAD

**Discussion by vabatta** [storybookjs/storybook#26891] [Help]
### Summary

Hi,

I've been trying to setup and work with web-components with a library such as [Stimulus](http://stimulus.hotwired.dev) or [Github Catalyst](https://catalyst.rocks) but I cannot figure out how to make the render. My components will be basically a js class that I need to import in an html template. Any suggestions?

**Comment by danipavic** [comment]
I had to resolve the same issue, basically what I had to do was:

- Create a Stimulus controller (I used the basic example from the Stimulus docs)
- Create a Story with a valid HTML
- In your `preview.js` file do the following:
   - Import Stimulus and controllers you are using
   - Invoke `start()` and `register()` functions
   
```
import { Application } from "https://unpkg.com/@hotwired/stimulus/dist/stimulus.js";
import HelloController from "../src/stories/hello_controller.js";

window.Stimulus = Application.start();

Stimulus.register("hello", HelloController);

```
https://github.com/user-attachments/assets/a2a69578-1e3c-40f2-86ff-57fd8f33eeeb
